velox/exec/tests/utils/LocalExchangeSource.cpp

```diff
@@ -130,11 +140,11 @@ class LocalExchangeSource : public exec::ExchangeSource {
           }
 
           if (!requestPromise.isFulfilled()) {
-            requestPromise.setValue();
+            requestPromise.setValue(Response{totalBytes, atEnd_});
```

Comment:
How did we tell before that there is no more data? Null or empty buffer?

Reply:
We didn't. ExchangeClient didn't need this information as it was simply going around the circle requesting data from sources.